internal/db/repos.go

```diff
@@ -520,11 +541,25 @@ func (s *RepoStore) List(ctx context.Context, opt ReposListOptions) (results []*
 		return nil, err
 	}
 
+	fromClause := sqlf.Sprintf("repo")
+	if opt.ExternalServiceID != 0 {
+		fromClause = sqlf.Sprintf("repo JOIN external_service_repos e ON repo.id = e.repo_id")
```

Comment:
You could actually push down the filter from the WHERE clause to here, which would simplify the code a bit and be as efficient (or more), I believe.

```suggestion
		fromClause = sqlf.Sprintf("repo JOIN external_service_repos e ON (repo.id = e.repo_id AND e.external_service_id) = %s", opt.ExternalServiceID)
```